packages/venia-concept/src/components/Main/main.js

```diff
@@ -8,21 +8,33 @@ import defaultClasses from './main.css';
 
 class Main extends Component {
     static propTypes = {
-        classes: PropTypes.shape({
-            root: PropTypes.string,
-            root_masked: PropTypes.string
+        classes: shape({
+            page: string,
+            page_masked: string,
+            root: string,
+            root_masked: string
         }),
-        isMasked: PropTypes.bool
+        isMasked: bool
     };
 
+    get classes() {
+        const { classes, isMasked } = this.props;
+        const suffix = isMasked ? '_masked' : '';
+
+        return ['page', 'root'].reduce(
+            (r, v) => ({ ...r, [v]: classes[`${v}${suffix}`] }),
```

Comment:
Small style change change: Rename the arguments from `r` to `accum` and `v` to `value`. Would make it a little easier to read.